builtin/game/falling.lua

```diff
@@ -146,13 +146,28 @@ end
 
 local function drop_attached_node(p)
 	local n = core.get_node(p)
+	local node_meta = core.get_meta(p)
+	local meta_fields = node_meta and node_meta:to_table().fields end
 	core.remove_node(p)
 	for _, item in pairs(core.get_node_drops(n, "")) do
 		local pos = {
 			x = p.x + math.random()/2 - 0.25,
 			y = p.y + math.random()/2 - 0.25,
 			z = p.z + math.random()/2 - 0.25,
 		}
+		-- If it drops itself, preserve its metadata.
+		if item == n and meta_fields then
```

Comment:
this doesn't make any sense since `item` will be a string while `n` is something like `{name="default:wood", param1=0, param2=0}`

Reply:
Actually, item will be a table as well, but you are correct in suggesting that it will not be equal.
This worked in 0.4.16, but now I should be dereferencing one level.